project_group_setup.py

```diff
@@ -257,11 +280,27 @@ def main(args):
         add_identifier_to_group(gid, type="osggroup", identifier_name=project_name)
         print(f"project {project_name}: added osggid {osggid_to_assign} and osg project name {project_name}")
 
+    for gid in projects_needing_cluster_groups:
+        request = {
+            "RequestType": "UnixClusterGroups",
+            "Version": "1.0",
+            "UnixClusterGroups": [{"Version": "1.0", "UnixClusterId": options.unix_id, "CoGroupId": gid}],
+        }
+        call_api3(
+            POST,
+            "unix_cluster/unix_cluster_groups.json",
+            request,
+        )
+        print(f"project group {gid}: added UNIX Cluster Group")
+
+    for project_gid in project_groups:
+        #Provision all project groups
+        call_api2(POST, f"co_provisioning_targets/provision/{options.provision_target}/cogroupid:{project_gid}.json")
```

Comment:
We should just need to provision projects that we find without cluster groups, right? If I'm reading this correctly, this will make a lot of calls for already provisioned groups

Reply:
I wanted to ensure that all non-provisioned projects with UNIX Cluster Groups got provisioned, even if this script wasn't the one to create it, but I couldn't find a way to determine which projects were or weren't provisioned already. If we accept that any projects that get partially set up by hand won't get provisioned we can just provision groups that the script has just made UNIX Cluster Groups for.

Reply:
> but I couldn't find a way to determine which projects were or weren't provisioned already.

If a group is in LDAP, it's been provisioned so you should be able to do a set difference vs the set of projects.

> If we accept that any projects that get partially set up by hand won't get provisioned we can just provision groups that the script has just made UNIX Cluster Groups for.

With the above, I don't think this quite matters but I think I'm ok with it.

Reply:
Added ldap search for unprovisioned groups at commit 085bb8d